logging/src/test/scala/com/ovoenergy/effect/LoggingTest.scala

```diff
@@ -0,0 +1,25 @@
+package com.ovoenergy.effect
+
+import cats.data.WriterT
+import cats.effect.IO
+import cats.instances.list._
+import cats.syntax.flatMap._
+import com.ovoenergy.effect.Logging.{Debug, Info, Log, _}
+import org.scalatest.{Matchers, WordSpec}
+
+class LoggingTest extends WordSpec with Matchers {
+
+  type LogWriter[A] = WriterT[IO, List[Log], A]
+  implicit val logging: Logging[LogWriter] = log => WriterT.tell(List(log))
+
+  "Logging syntax" should {
+
+    "Log before running the actual effect" in {
+      val message = Debug("Testing logging")
+      val toLog: LogWriter[Unit] = WriterT.liftF(IO.unit)
+      val expected = List(message, Info("boo"))
+      val events = (toLog >> Logging[LogWriter].log(Info("boo"))).log(message)
+      events.written.unsafeRunSync shouldEqual expected
+    }
```

Comment:
Shouldn't `expected` be `List(Info("boo"), message)` since at line 21 `log.info` is called first?

Reply:
ah no, the use case of this `.log` syntax is to be able to easily _prepend_ logs to effects

e.g.

```scala
http.get("http://example.com").log("Requesting examples")
```

will still log `Requesting examples` even if the http call returns a failed effect